From b1d12ec37208fe92044a134e6f7bbde274f185cd Mon Sep 17 00:00:00 2001
From: Fugang Duan <B38611@freescale.com>
Date: Mon, 28 May 2012 10:58:19 +0800
Subject: ENGR00211129 - SPDC : modify waveform update modes.

- Because mode_4 only use for handwriting mode, and
  mode_3 use for power off mode, modify the waveform
  mode for update requests.
- Use mode_2 for DU and A2, mode_1 for GU:
  mode_A2 = mode_2, mode_du = mode_2;
  mode_gu = mode_1;

Signed-off-by: Fugang Duan  <B38611@freescale.com>
---
 drivers/video/mxc/mxc_spdc_fb.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/video/mxc/mxc_spdc_fb.c b/drivers/video/mxc/mxc_spdc_fb.c
index 37938b9..1659724 100644
--- a/drivers/video/mxc/mxc_spdc_fb.c
+++ b/drivers/video/mxc/mxc_spdc_fb.c
@@ -3267,7 +3267,7 @@ static ssize_t store_update(struct device *device,
 	else if (strncmp(buf, "init", 4) == 0)
 		update.waveform_mode = fb_data->wv_modes.mode_init;
 	else if (strncmp(buf, "gu", 2) == 0)
-		update.waveform_mode = SPDC_WAV_MODE_3;
+		update.waveform_mode = SPDC_WAV_MODE_1;
 	else if (strncmp(buf, "auto", 4) == 0)
 		update.waveform_mode = WAVEFORM_MODE_AUTO;
 
@@ -3774,7 +3774,7 @@ static int __devinit mxc_spdc_fb_probe(struct platform_device *pdev)
 	 * Should be overwritten via ioctl.
 	 */
 	fb_data->wv_modes.mode_init = SPDC_WAV_MODE_DEFAULT;
-	fb_data->wv_modes.mode_du = SPDC_WAV_MODE_4;
+	fb_data->wv_modes.mode_du = SPDC_WAV_MODE_2;
 	fb_data->wv_modes.mode_gc4 = SPDC_WAV_MODE_2;
 	fb_data->wv_modes.mode_gc8 = SPDC_WAV_MODE_1;
 	fb_data->wv_modes.mode_gc16 = SPDC_WAV_MODE_1;
-- 
1.7.9.5

